homeassistant/components/vizio/config_flow.py

```diff
@@ -90,7 +92,11 @@ def _get_pairing_schema(input_dict: Dict[str, Any] = None) -> vol.Schema:
 
 def _host_is_same(host1: str, host2: str) -> bool:
```

Comment:
This now has to run in the executor thread pool as there's I/O inside.

Reply:
Good point. So now this function has to be async I think? If so, how do I handle the path where socket.gethostbyname is never called and there is therefore nothing to block on?

Reply:
Good point. So now this function has to be async I think? If so, how do I handle the path where socket.gethostbyname is never called and there is therefore nothing to block on?

Reply:
Good point. So now this function has to be async I think? If so, how do I handle the path where socket.gethostbyname is never called and there is therefore nothing to block on?

Reply:
‘await asyncio.sleep(0)’ maybe?

Reply:
I'd schedule this whole function on the executor from the steps where it's needed.

Reply:
better idea 👍